### PR TITLE
Report the repository name when sending email errors.

### PR DESCRIPTION
**Pull request by ltratt**
At the moment, if a job fails users get an email with a subject "snare error" --
it can then be a bit difficult to work out what repository the error in the
email body refers to. This commit changes it so that the email subject "snare
error when running <repo>".

### CODE REVIEW

File: src/jobrunner.rs [hunk 1]
```diff
@@ -479,10 +479,10 @@ impl JobRunner {
     }
 
     /// If the user has specified an email address, send the contents of
-    fn sendemail(&self, email: &Option<String>, mut file: &File) {
-        if let Some(ref toaddr) = email {
+    fn sendemail(&self, job: &Job, mut file: &File) {
+        if let Some(ref toaddr) = job.rconf.email {
             let mut buf = Vec::new();
-            buf.extend(b"Subject: snare error\n\n");
+            buf.extend(format!("Subject: snare error when running {}\n\n", job.repo_id).as_bytes());
```
Comment (vext01) [inline src/jobrunner.rs]: Does it make sense to "run" a repo?

How about something like:
```
[snare] encountered errors for repo "wibble".
```

**Comment by vext01** [comment]
one comment

File: src/jobrunner.rs (hunk 1 above)
Comment (ltratt) [reply src/jobrunner.rs]: Agreed that something else would be better. Try https://github.com/softdevteam/snare/pull/52/commits/feebdf968eb0a59a6401530aa982ec143c37ddd0.

**Comment by vext01** [comment]
LGTM. please squash

**Comment by ltratt** [comment]
Squashed.

**Comment by vext01** [comment]
That's unfortunate:
```
error[E0433]: failed to resolve: could not find `__rt` in `quote`
   --> /home/buildbot-worker5/worker5/buildbot-build-script/build/.cargo/registry/src/github.com-1ecc6299db9ec823/failure_derive-0.1.6/src/lib.rs:107:70
    |
107 | fn display_body(s: &synstructure::Structure) -> Result<Option<quote::__rt::TokenStream>, Error> {
    |                                                                      ^^^^ could not find `__rt` in `quote`
error: aborting due to previous error
```

**Comment by ltratt** [comment]
That will be fixed shortly and then we can retry https://github.com/rust-lang-nursery/failure/issues/342

**Comment by ltratt** [comment]
@vext01 You should be able to retry this now.

**Pull request merged**